pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTBlockStatement.java

```diff
@@ -34,8 +36,10 @@ protected void handleSourceCode(final String source) {
         // check, whether the this block statement really begins with a curly brace
         // unfortunately, for-loop and if-statements always contain a block statement,
         // regardless whether curly braces where present or not.
-        // char firstChar = source.charAt(node.getLoc().getStartIndex());
-        // curlyBrace = firstChar == '{';
-        // TODO(b/239648780)
+        SourceLocation loc = node.getSourceLocation();
+        String[] lines = source.split("\n");
```

Comment:
suggest one or more helper function here. Maybe one useful one would be `SourceLocation.extractFrom(String source)String`